flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/plan/metadata/FlinkMetadata.java

```diff
@@ -238,4 +239,20 @@ ImmutableBitSet getUniqueGroups(
             RelWindowProperties getWindowProperties(RelNode r, RelMetadataQuery mq);
         }
     }
+
+    /** Metadata about which combinations of columns are change log upsert identifiers. */
+    public interface ChangeLogUpsertKeys extends Metadata {
```

Comment:
how about name it ChangeLogDistributionKey?  I think it would be closer to what the key describes

Reply:
I think upsert is more intuitive.
Because:
- the downstream operators upsert optimization needs this.
- the downstream upsert sink needs this.

Reply:
I am considering just name it `UpsertKey`....